inst/report/safetyGraphicsReport.Rmd

```diff
@@ -0,0 +1,171 @@
+---
+output:
+  html_document
+
+params:
+  data: NA
+  mapping: NA
+  charts: NA
+  
+  
+---
+
+## Customized Interactive Safety Graphics {.tabset .tabset-fade} 
+
+```{r echo = FALSE}
+
+# Function to create chart-level params
+create_chart_params <- function(data, chart, mapping){
+  #convert settings from data frame to list and subset to specified domain (if any)
+  settingsList <-  safetyGraphics::generateMappingList(mapping, domain=chart$domain)
+  
+  #subset data to specific domain (if specified)
+  if(chart$domain=="multiple"){
+    domainData <- data
+  }else{
+    domainData<- data[[chart$domain]]
+  }
+  params <- list(data=domainData, settings=settingsList)
+
+  return(params)
+}
+
+# Function to create chart
+create_chart <- function(chart, params){
+  if (chart$type=="htmlwidget"){
```

Comment:
Pretty sure we need to add the init function (if any) in to the workflow here. I'll see what I can do in the next few days.